src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -86,16 +82,17 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
 
 		            //BEFORE INSERT - we do it in the Before so we don't get the affiliation we just created when we query for
 		            //affls of the same type.
-		            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+		            if (triggerAction == TDTM_Runnable.Action.BeforeInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before_Insert)) {
 		                if (affl.Primary__c && affl.Contact__c != null && affl.Account__c != null) {
 		            	    afflsMadePrimary.add(affl);
 		            	    afflMadePrimary = true;
 		            	    populateKeyAffls(affl, lookupFieldName, relatedContact, dmlWrapper);
 		                }
+                        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before_Insert, true);
```

Comment:
@xinnanlisf If I'm reading this correctly, this line, in combination with line 85 is only allowing the Contact's primary affiliation field to get set for the first Affiliation record in newList...we're setting checking within the newList's for loop.